src/qubex/analysis/fitting.py

```
@@ -1329,10 +1642,337 @@ def residuals(params, f, y):
+        "kappa_in": kappa_in,
+        "A": A,
+        "phi": phi,
+        "fig": fig,
+    }
+
 
+def fit_rotation(
+    times: npt.NDArray[np.float64],
+    data: npt.NDArray[np.float64],
+    offresonant: bool = False,
+    r0: npt.NDArray[np.float64] = np.array([0, 0, 1]),
```

Ruff (0.8.2)

Do not perform function call `np.array` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)